### PR TITLE
[GraphBolt] Get feature keys of FeatureStore.

### PR DESCRIPTION
**Pull request by yxy235**
## Description

Get feature keys of FeatureStore.
API: FeatureStore.keys()
return: List[tuple(domain, type_name, feat_name)]

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: tests/python/pytorch/graphbolt/impl/test_torch_based_feature_store.py [hunk 1]
```diff
@@ -239,6 +239,13 @@ def test_torch_based_feature_store(in_memory):
             "edge", "paper:cites:paper", "b"
         ) == torch.Size([2, 2])
 
+        # Test get the keys of the features.
+        # assert 0, print(feature_store.keys())
```
Comment (Skeleton003) [inline tests/python/pytorch/graphbolt/impl/test_torch_based_feature_store.py:243]: remove it

File: python/dgl/graphbolt/feature_store.py [hunk 2]
```diff
@@ -171,3 +171,13 @@ def update(
...
+
+    def keys(self):
+        """Get the keys of the features.
+
+        Returns
+        -------
+        List[tuple]
+            The keys of the features.
+        """
+        return []
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/feature_store.py:183]: `raise NotImplementedError` to keep consistent with other APIs.

File: python/dgl/graphbolt/feature_store.py [hunk 3]
```diff
@@ -171,3 +171,13 @@ def update(
...
         """
         raise NotImplementedError
+
+    def keys(self):
+        """Get the keys of the features.
+
+        Returns
+        -------
+        List[tuple]
+            The keys of the features.
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/feature_store.py:181]: more specific about what kind of tuple, it's (domain, ntype/etype, feat_name)

File: python/dgl/graphbolt/impl/basic_feature_store.py [hunk 4]
```diff
@@ -136,3 +136,13 @@ def update(
...
         """Return the number of features."""
         return len(self._features)
+
+    def keys(self):
+        """Get the keys of the features.
+
+        Returns
+        -------
+        List[tuple]
+            The keys of the features.
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/impl/basic_feature_store.py:146]: ditto

**Review by Rhett-Ying — APPROVED** [review]

**Pull request merged**